crates/sui-core/src/authority/authority_store.rs

```diff
@@ -354,6 +354,31 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
         Ok(result)
     }
 
+    /// Get many objects by their (id, version number) key.
+    pub fn get_sequenced_input_objects(
+        &self,
+        digest: &TransactionDigest,
+        objects: &[InputObjectKind],
+    ) -> Result<Vec<Option<Object>>, SuiError> {
+        let shared_locks: HashMap<_, _> = self.all_shared_locks(digest)?.into_iter().collect();
+
+        let mut result = Vec::new();
+        for kind in objects {
+            let obj = match kind {
+                InputObjectKind::MovePackage(id) => self.get_object(id)?,
+                InputObjectKind::SharedMoveObject(id) => match shared_locks.get(id) {
+                    Some(version) => self.get_object_by_key(id, *version)?,
+                    None => None,
```

Comment:
I think this should be something like `shared_locks.get(id).expect("Did not find shared lock for shared object")` right?
Since this is basically an invariant violation, we should not just return None